batch/batch/front_end/front_end.py

```diff
@@ -500,9 +500,10 @@
 ''',
                                           jobs_args)
                 except pymysql.err.IntegrityError as err:
-                    if err.args[1] == 1022:
+                    if err.args[1] in (1022, 1062, 1586, 1859, 3026):
```

Comment:
Can I get a comment so I can interpret these?

Reply:
Looks like:

1022 Can't write; duplicate key in table '%s'
1062 Duplicate entry '%s' for key %d
1586 Duplicate entry '%s' for key '%s'
1859 Duplicate entry for key '%s'
3026 Duplicate entry '%s'.

How did you generate this list?  I'm slightly confused why there are so many errors for the same thing, and why one would be generated vs another.

1022 is a metadata error: "MySQL Error 1022 occurs when we try to create a foreign key constraint with a name as already existing constraint name within the same database schema."

3026 ER_DUP_LIST_ENTRY appears to be a duplicate systems variable entry.

I don't see why it would generate anything besides 1062:

```
mysql> create table `duptest` (x INT, y INT, z INT, PRIMARY KEY(x, y));
Query OK, 0 rows affected (0.02 sec)

mysql> insert into duptest values (1, 2, 3);
Query OK, 1 row affected (0.00 sec)

mysql> insert into duptest values (1, 2, 3);
ERROR 1062 (23000): Duplicate entry '1-2' for key 'PRIMARY'
```

I get the same error for a unique constraint and for a non-compound key.

I vote for only checking 1062 Duplicate entry '%s' for key %d.

Reply:
I searched the mysql error list for "duplicate" and selected anything that seemed likely related to a duplicate key.

Where did you get the detailed explanation of that error? The official MySQL docs don't give any further information.

I don't see any information that says that 1586 isn't a reasonable error to report on a duplicate entry for a key, in fact 1062 indicates it uses the format string from 1586! 1859 also seems sufficiently vague as to indicate the same error.

I guess we can put 1062 in for now and wait for other errors to occur.